internal/dataplane/fallback/fallback.go

```diff
@@ -37,7 +37,7 @@ func (g *Generator) GenerateExcludingBrokenObjects(
 		return store.CacheStores{}, fmt.Errorf("failed to build cache graph: %w", err)
 	}
 
-	fallbackCache, _, err := cache.TakeSnapshotIfChanged(store.SnapshotHashEmpty)
+	fallbackCache, err := cache.TakeSnapshot()
```

Comment:
It might make sense to keep `TakeSnapshot` as we have a use case for it here. In this case, we unconditionally want to make a copy of the cache, so there's no point in calculating the hash sum if we know this will have no use upfront.

Reply:
Good point, two methods are justifiable